File: src/#821/solution.js
var shortestToChar = function (S, C) {
  let Cindex = []
  S = S.split('')
  S.forEach((v, i) => {
    if (C === v) {
      Cindex.push(i)
    }
  })
  let res = []
  S.forEach((v, i) => {
    if (v === C) {
      res.push(0)
    } else {
      let min = 10000
      Cindex.forEach(v => {
        let dis = i - v
        if (dis < 0) {
          dis *= -1
        }
        if (dis < min) {
          min = dis
        }
      })
      res.push(min)
    }
  })
  return res
}

File: src/#821/solution.go
package solution

func shortestToChar(S string, C byte) []int {
	Cindex := []int{}
	for i, v := range []byte(S) {
		if v == C {
			Cindex = append(Cindex, i)
		}
	}
	res := []int{}
	for i, v := range []byte(S) {
		if v == C {
			res = append(res, 0)
		} else {
			min := 10000
			for _, idx := range Cindex {
				dis := i - idx
				if dis < 0 {
					dis *= -1
				}
				if dis < min {
					min = dis
				}
			}
			res = append(res, min)

		}
	}
	return res
}
